dapp/tests/unit/validation.test.js

```diff
@@ -0,0 +1,183 @@
+const {
+  validateWorkerEnv,
+  validateAppSecret,
+  validateRequesterSecret,
+  validateProtectedData,
+} = require('../../src/validation');
+
+describe('validateWorkerEnv function', () => {
+  let testedObj;
+
+  beforeEach(() => {
+    testedObj = {
+      IEXEC_OUT: '/path/to/iexec/out',
+    };
+  });
+
+  it('should not throw an error if all input values are valid', () => {
+    expect(() => validateWorkerEnv(testedObj)).not.toThrow();
+  });
+
+  it('should throw an error if any required input is missing', () => {
+    delete testedObj.IEXEC_OUT;
+    expect(() => validateWorkerEnv(testedObj)).toThrow(
+      Error('Worker environment error: "IEXEC_OUT" is required')
+    );
+  });
```

Comment:
It's been a while I haven't seen a `delete` keyword!

Not sure the `beforeEach` is really necessary here?

```suggestion
  it('should not throw an error if all input values are valid', () => {
    expect(() =>
      validateWorkerEnv({
        IEXEC_OUT: '/path/to/iexec/out',
      })
    ).not.toThrow();
  });

  it('should throw an error if any required input is missing', () => {
    expect(() => validateWorkerEnv({})).toThrow(
      Error('Worker environment error: "IEXEC_OUT" is required')
    );
  });
```

Reply:
changed fc0e83ac6ffbd1aff60db793330fe5556d832f94